Sources/ApexyURLSession/URLSession+AsyncAwait13.swift

```diff
@@ -13,65 +13,54 @@ import Foundation
 @available(watchOS, introduced: 6, deprecated: 8, message: "Extension is no longer necessary. Use API built into SDK")
 @available(tvOS, introduced: 13, deprecated: 15, message: "Extension is no longer necessary. Use API built into SDK")
 extension URLSession {
-
-    typealias ContentContinuation = CheckedContinuation<(Data, URLResponse), Error>
-    
-    private func adaptToAsync(
-        dataTaskClosure: (ContentContinuation) -> URLSessionDataTask) async throws -> (Data, URLResponse) {
-            let progressWrapper = ProgressWrapper()
-            return try await withTaskCancellationHandler(handler: {
-                progressWrapper.cancel()
-            }, operation: {
-                try await withCheckedThrowingContinuation { (continuation: ContentContinuation) in
-                    let task = dataTaskClosure(continuation)
-                    progressWrapper.progress = task.progress
-                    
-                    task.resume()
-                }
-            })
-    }
     
     public func data(
         for request: URLRequest,
         delegate: URLSessionTaskDelegate? = nil) async throws -> (Data, URLResponse) {
-            return try await adaptToAsync(dataTaskClosure: { continuation in
-                return self.dataTask(with: request) { data, response, error in
+            return try await AsyncAwaitHelper.adaptToAsync(dataTaskClosure: { continuation in
+                let task = self.dataTask(with: request) { data, response, error in
```

Comment:
No memory leaks here?

Reply:
i removed `self`. i think there is no leak now